src/build.rs

```diff
@@ -251,21 +288,34 @@ fn load_manifest_overrides(root_path: &PathBuf) -> anyhow::Result<Overrides> {
 
     let toml = manifest_str.parse::<toml::Value>()?;
 
-    let meta = toml
+    let capi = toml
         .get("package")
         .and_then(|v| v.get("metadata"))
         .and_then(|v| v.get("capi"));
 
-    Ok(if let Some(meta) = meta {
-        Overrides {
-            header_name: meta
-                .get("header_name")
+    let header = capi.and_then(|v| v.get("header"));
+
+    let header = if let (Some(ref capi), Some(ref header)) = (capi.as_ref(), header.as_ref()) {
+        HeaderOverrides {
+            name: header
+                .get("name")
+                .or_else(|| capi.get("header_name"))
                 .map(|v| v.clone().try_into())
                 .unwrap_or(Ok(None))?,
```

Comment:
It's not added but whatever extension is there already is replaced with ".h" like before, and if you don't provide an extension then it's added :) That behaviour didn't change compared to before, that's from the 3 `set_extension("h")` calls.

Reply:
that part is fine, but the code is not picking the name as you see with nnnoiseless (that's the only crate using the feature)

Reply:
That was the case before but with the latest version here it seems to work fine for me. Problem was that it was going into the default case if the `capi.header` configuration did not exist at all, instead of using the fallback "header_name" from the `capi` configuration.